eng/common/scripts/TypeSpec-Project-Generate.ps1

```diff
@@ -46,14 +46,26 @@ function NpmInstallForProject([string]$workingDirectory) {
         Write-Host("Copying package.json from $replacementPackageJson")
         Copy-Item -Path $replacementPackageJson -Destination "package.json" -Force
 
+        #default to root/eng/emitter-package-lock.json but you can override by writing
+        #Get-${Language}-EmitterPackageLockPath in your Language-Settings.ps1
+        $emitterPackageLock = Join-Path $PSScriptRoot "../../emitter-package-lock.json"
+        if (Test-Path "Function:$GetEmitterPackageLockPathFn") {
+            $emitterPackageLock = &$GetEmitterPackageLockPathFn
+        }
+
+        if (Test-Path $emitterPackageLock) {
+            Write-Host("Copying package-lock.json from $emitterPackageLock")
+            Copy-Item -Path $emitterPackageLock -Destination "package-lock.json" -Force
+        }
+
         $useAlphaNpmRegistry = (Get-Content $replacementPackageJson -Raw).Contains("-alpha.")
 
         if($useAlphaNpmRegistry) {
             Write-Host "Package.json contains '-alpha.' in the version, Creating .npmrc using public/azure-sdk-for-js-test-autorest feed."
-            "registry=https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-js-test-autorest/npm/registry/ `n`nalways-auth=true" | Out-File '.npmrc'
+            "registry=https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-js-test-autorest@local/npm/registry/ `n`nalways-auth=true" | Out-File '.npmrc'
         }
 
-        npm install --no-lock-file
+        npm install
```

Comment:
I think we should use `npm ci` if there is a lock file, and `npm install` if not.

Reply:
Updated to conditionally use `npm ci` or `npm install`